neuralop/layers/codano.py

```diff
@@ -0,0 +1,607 @@
+        fno_skip (str): Type of skip connection to be used, by default 'linear'.
+        mlp_skip (str):  module to use for ChannelMLP skip connections, by default "soft-gating".
+        channel_mlp_expansion (float): expansion parameter for self.channel_mlp, by default 0.5.
+        separable (bool): whether to use separable convolutions, by default False.
+        factorization (str): Type of factorization to be used, by default 'tucker'.
+        rank (float): Rank of the factorization, by default 1.0.
+        SpectralConvolution (callable): Spectral convolution module to be used.
+        joint_factorization (bool): whether to factorize all spectralConv weights as one tensor, by default False
+
+        # Normalization
+        Normalizer (callable): Normalization module to be used.
+
+        codimension_size (int): Size of the codimension for whole function, only used for permutation_eq = False.
+        per_channel_attention (bool): whether to use per channel attention, by default True (overwrite token_codimension to 1).
+        permutation_eq (bool): whether to use permutation equivariant mixer layer  after attention mechanism.
+        temperature (float): Temperature parameter for attention mechanism.
+        kqv_non_linear (bool): whether to use non-linear activation for K,Q,V operator.
+    """
+
+    def __init__(
+        self,
+        n_modes,
+        n_head=1,
+        token_codimension=1,
+        output_scaling_factor=None,
+        incremental_n_modes=None,
+        head_codimension=None,
+        use_channel_mlp=False,
+        non_linearity=F.gelu,
+        preactivation=False,
+        fno_skip='linear',
+        channel_mlp_skip='soft-gating',
+        channel_mlp_expansion=1.0,
+        separable=False,
+        factorization='tucker',
+        rank=1.0,
+        SpectralConvolution=None,
+        Normalizer=None,
+        joint_factorization=False,
+        fixed_rank_modes=False,
+        implementation='factorized',
+        decomposition_kwargs=None,
+        fft_norm='forward',
+        codimension_size=None,
+        per_channel_attention=True,
+        permutation_eq=True,
+        temperature=1.0,
+        kqv_non_linear=False,
+        **_kwargs,
+    ):
+        super().__init__()
+
+        # Co-dimension of each variable/token. The token embedding space is
+        # identical to the variable space, so their dimensionalities are equal.
+        self.token_codimension = token_codimension
+
+        # (maybe) codim of attention from each head
+        self.head_codimension = (head_codimension
+                                 if head_codimension is not None
+                                 else token_codimension)
+        self.n_head = n_head  # number of heads
+        self.output_scaling_factor = output_scaling_factor
+        self.temperature = temperature
+
+        # K,Q,V operator with or without non_lin
+
+        if kqv_non_linear:
```

Comment:
is there a simpler thing we could call this? Is `kqv` the same as attention?

Reply:
Yes, the notation aligns with the regular attention mechanism.

Reply:
So could we call this `nonlinear_attention` or similar?